src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -237,11 +237,19 @@ private void GenerateProlog()
             LLVMBasicBlockRef prologBlock = _llvmFunction.AppendBasicBlock("Prolog");
             _builder.PositionAtEnd(prologBlock);
 
-
-            // Allocate a slot to store exceptions being dispatched
+            // Allocate slots to store exception being dispatched and generic context if present
             if(_exceptionRegions.Length > 0)
             {
                 _spilledExpressions.Add(new SpilledExpressionEntry(StackValueKind.ObjRef, "ExceptionSlot", GetWellKnownType(WellKnownType.Object), 0, this));
+                // and a slot for the generic context if present
+                if (FuncletsRequireHiddenContext())
```

Comment:
I think you should be able change implementation of `FuncletsRequireHiddenContext` to be just `_method.RequiresInstArg()`. The current implementation looks too complex for no good reason.

Reply:
Thanks, yes that works, done.